testsuite/MDAnalysisTests/auxiliary/base.py

```diff
@@ -328,29 +328,34 @@ def test_get_auxreader_for(self, ref, reader):
         reader = mda.auxiliary.core.get_auxreader_for(ref.testdata)
         assert reader == ref.reader
 
-    def test_iterate_through_trajectory(self, ref):
-        # add to trajectory
-        u = mda.Universe(COORDINATES_TOPOLOGY, COORDINATES_XTC)
-        u.trajectory.add_auxiliary('test', ref.testdata)
+    def test_iterate_through_trajectory(self, ref, ref_universe):
         # check the representative values of aux for each frame are as expected
         # trajectory here has same dt, offset; so there's a direct correspondence
         # between frames and steps
-        for i, ts in enumerate(u.trajectory):
+        for i, ts in enumerate(ref_universe.trajectory):
             assert_equal(ts.aux.test, ref.auxsteps[i].data,
-                         "representative value does not match when iterating through all trajectory timesteps")
-        u.trajectory.close()
+                         "representative value does not match when "
+                         "iterating through all trajectory timesteps")
 
-    def test_iterate_as_auxiliary_from_trajectory(self, ref):
-        # add to trajectory
-        u = mda.Universe(COORDINATES_TOPOLOGY, COORDINATES_XTC)
-        u.trajectory.add_auxiliary('test', ref.testdata)
+    def test_iterate_as_auxiliary_from_trajectory(self, ref, ref_universe):
         # check representative values of aux for each frame are as expected
         # trahectory here has same dt, offset, so there's a direct correspondence
         # between frames and steps, and iter_as_aux will run through all frames
-        for i, ts in enumerate(u.trajectory.iter_as_aux('test')):
+        for i, ts in enumerate(ref_universe.trajectory.iter_as_aux('test')):
             assert_equal(ts.aux.test, ref.auxsteps[i].data,
-                         "representative value does not match when iterating through all trajectory timesteps")
-        u.trajectory.close()
+                         "representative value does not match when "
+                         "iterating through all trajectory timesteps")
+
+    def test_auxiliary_read_ts_rewind(self, ref_universe):
+        # AuxiliaryBase.read_ts() should retrieve the correct step after
+        # reading the last one. Issue #2674 describes a case in which the
+        # object gets stuck on the last frame.
+        aux_info_0 = ref_universe.trajectory[0].aux.test
+        ref_universe.trajectory[-1].aux.test
```

Comment:
I think just `ref_universe.trajectory[-1]` should work here?

Reply:
You're right! Just to be sure I've checked, and the test indeed fails with just `ref_universe.trajectory[-1]` without the fix.